lib/AST/ASTContext.cpp

```diff
@@ -1839,28 +1842,14 @@ void ASTContext::addLoadedModule(ModuleDecl *M) {
   getImpl().LoadedModules[M->getName()] = M;
 }
 
-static AllocationArena getArena(GenericSignature genericSig) {
-  if (!genericSig)
-    return AllocationArena::Permanent;
-
-  if (genericSig.hasTypeVariable()) {
-    assert(false && "Unsubstituted type variable leaked into generic signature");
```

Comment:
Do we preserve this assert anywhere else?

Reply:
Yep, GenericSignature::get() on ASTContext.cpp line 4476. That seems to be the only entry point where we actually create new signatures, so I thought that was the best place to put it.

Reply:
I actually added that assert recently in 02db6324888bfb77a619b8f2433721e2f7ba0bf9, but kept the remaining logic around until I could do more tests. You re-worded it in 74b653518df3ac5304229b31b5ef0d2eb5f3db0f :-)